File: packages/types/src/ai.ts
export type ProviderName = 'anthropic' | 'openai' | 'gemini' | 'nvidia' | 'ollama' | 'groq';

export type TaskTier = 'fast' | 'balanced' | 'powerful';

export interface ProviderConfig {
  name: ProviderName;
  enabled: boolean;
  hasApiKey: boolean;
  fastModel: string;
  balancedModel: string;
  powerfulModel: string;
  baseUrl?: string;
}

export interface ModelInfo {
  id: string;
  display_name: string;
  supports_tools: boolean;
  max_tokens: number | null;
  context_window: number | null;
}

File: crates/aurora-core/src/types/ai.rs
use serde::{Deserialize, Serialize};

/// model tier - determines which model is selected from
/// the active provider's tier mapping.
#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
pub enum TaskTier {
    Fast,
    Balanced,
    Powerful,
}

/// Provider name identifier.
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum ProviderName {
    Anthropic,
    OpenAi,
    Gemini,
    Nvidia,
    Ollama,
    Groq,
}

/// A single message in an AI conversation.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AiMessage {
    pub role: String, // "user" | "assistant" | "system"
    pub content: String,
}

/// Event payload emitted during AI streaming.
#[derive(Debug, Clone, Serialize)]
pub struct AIStreamChunkEvent {
    pub request_id: String,
    pub chunk: String,
    pub done: bool,
}

/// Model metadata returned by provider model-listing APIs.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ModelInfo {
    pub id: String,
    pub display_name: String,
    pub supports_tools: bool,
    pub max_tokens: Option<u32>,
    pub context_window: Option<u32>,
}
